HW4_2/supl_files/dna_rna_modules.py

```diff
@@ -0,0 +1,103 @@
+REPLACE_VALUES: dict[str, str] = {"t": "u", "T": "U"}
+COMPLEMENT_VALUES: dict[str, str] = {
+    "A": "T",
+    "a": "t",
+    "G": "C",
+    "g": "c",
+    "U": "A",
+    "u": "a",
+    "T": "A",
+    "t": "a",
+    "C": "G",
+    "c": "g",
+}
+nucleotides: list[str] = ["A", "a", "T", "t", "G", "g", "C", "c", "U", "u"]
+
+
+def transcription(*sequence: list[str], replace_values=REPLACE_VALUES) -> list:
+    result = []
+    for seq in sequence:
+        for previous, new in replace_values.items():
+            seq = seq.replace(previous, new)
+        result.append(seq)
+    return result
+
+
+def complementation(*sequence: list[str], complement_values=COMPLEMENT_VALUES) -> list:
+    result = []
+    translation_table = str.maketrans(complement_values)
+    for seq in sequence:
+        seq = seq.translate(translation_table)
+        result.append(seq)
+    return result
+
+
+def reversion(*sequence: list[str]) -> list:
+    result = []
+    for seq in sequence:
+        seq = seq[::-1]
+        result.append(seq)
+    return result
+
+
+def reverse_compl(*sequence: list[str]) -> list:
+    rev_seq = reversion(*sequence)
+    result = []
+    for seq in rev_seq:
+        rev_compl = complementation(seq)
+        result.extend(rev_compl)
+    return result

```

Comment:
```suggestion
    rev_seq = reversion(*sequence)
    result = complementation(*rev_seq )
    return result
```

А так не срааботает?
В любом случае - у тебя везде получается итерирование по сиквенсам копируется. Лучше тогда было бы вынести это один раз в главную функцию